File: 3-20/miaoz/1.cpp
class Solution {
public:
    vector<int> getLeastNumbers(vector<int>& arr, int k) {
        priority_queue<int,vector<int> ,less<int> > q;
        for(int i=0;i<arr.size();i++){
            q.push(arr[i]);
            while(q.size()>k)q.pop();
        }
        vector<int> res;
        while(!q.empty()){
            res.push_back(q.top());
            q.pop();
        }
        return res;
    }
};

File: 3-20/miaoz/1.java
class Solution {
    public int[] getLeastNumbers(int[] arr, int k) {
        Queue q=new PriorityQueue<Integer>((a,b)->{return b-a;});
        for(int i=0;i<arr.length;i++){
            q.offer(arr[i]);
            while(q.size()>k)q.poll();
        }
        int []res=new int[k];
        for(int i=0;i<k;i++){
            res[i]=(int)q.peek();
            q.poll();
        }
        return res;
    }
}